core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -266,62 +265,68 @@ export class RteHelperModalComponent {
+            'the end time.');
+        } else {
+          this.updateRteErrorMessage('');
+        }
       }
     } else if (this.componentId === this.COMPONENT_ID_LINK) {
       let url: string = value[0];
       let text: string = value[1];
       if (text === '') {
         text = url;
-        this.saveButtonIsDisabled = false;
-      }
-      // First check if the `text` looks like a URL.
-      const suffixes = ['.com', '.org', '.edu', '.gov'];
-      let textLooksLikeUrl = suffixes.some(suffix => text.endsWith(suffix));
-      if (!textLooksLikeUrl) {
-        this.saveButtonIsDisabled = false;
+        this.updateRteErrorMessage('');
       } else {
-      // If the text looks like a URL, strip the leading 'http://' or
-      // 'https://' or 'www.'.
-        const prefixes = ['https://', 'http://', 'www.'];
-        if (prefixes.some(prefix => url.startsWith(prefix))) {
-          const urlPrefix = prefixes.find(prefix => url.startsWith(prefix));
-          url = url.substring(urlPrefix.length);
-        }
-        if (prefixes.some(prefix => text.startsWith(prefix))) {
-          const textPrefix = prefixes.find(prefix => text.startsWith(prefix));
-          text = text.substring(textPrefix.length);
+        // First check if the `text` looks like a URL.
+        const suffixes = ['.com', '.org', '.edu', '.gov'];
+        let textLooksLikeUrl = suffixes.some(suffix => text.endsWith(suffix));
+        if (!textLooksLikeUrl) {
+          this.updateRteErrorMessage('');
+        } else {
+          // If the text looks like a URL, strip the leading 'http://' or
+          // 'https://' or 'www.'.
+          const prefixes = ['https://', 'http://', 'www.'];
+          for (const prefix of prefixes) {
+            if (url.startsWith(prefix)) {
+              url = url.substring(prefix.length);
+            }
+            if (text.startsWith(prefix)) {
+              text = text.substring(prefix.length);
+            }
+          }
+          // After the cleanup, if the strings are not equal, then we do not
+          // allow the lesson creator to save it.
+          if (url !== text) {
+            this.updateRteErrorMessage(
+              'It seems like clicking on this link will lead the user to a ' +
+              'different URL than the text specifies. Please change the text.'
+            );
+          } else {
+            this.updateRteErrorMessage('');
+          }
         }
-        // After the cleanup, if the strings are not equal, then we do not
-        // allow the lesson creator to save it.
-        this.saveButtonIsDisabled = (url !== text);
-      }
-      if (this.saveButtonIsDisabled) {
-        this.updateRteErrorMessage(
-          'It seems like clicking on this link will lead the user to a ' +
-          'different URL than the text specifies. Please change the text.'
-        );
-      } else {
-        this.updateRteErrorMessage('');
       }
     }
   }
 
+  isErrorMessageNonEmpty(): boolean {
```

Comment:
NonEmpty --> Nonempty

Reply:
done in the [PR](https://github.com/oppia/oppia/pull/20071)